acme/commons.go

```diff
@@ -200,6 +200,14 @@ type Order struct {
 	Replaces string `json:"replaces,omitempty"`
 }
 
+func (r *Order) Err() error {
+	if r.Error != nil {
```

Comment:
This method doesn't do anything useful. The expression `r.Error != nil ? r.Error : nil` can be condensed to just `r.Error`.

Do you want to add a nil check on `r`, i.e.

```suggestion
	if r != nil {
}
```

which would avoid a panic? https://go.dev/play/p/e9z35S2ydbu

Reply:
You missed the point: the panic doesn't happen because the order is nil, but because `Error` is nil.

> This method doesn't do anything useful.

This method is doing something useful.

Reply:
>> This method doesn't do anything useful.
>
> This method is doing something useful.

It's not being used though, how can it be useful?

I have locally removed the `(*Order) Err()` and `(*Challenge) Err()` methods, and tests still pass.

> https://go.dev/play/p/9X_m55Un5fI

That's surprising :thinking:

Reply:
>>  I have locally removed the (*Order) Err() and (*Challenge) Err() methods, and tests still pass.
>
> I added a new commit yesterday, so yes, now they are unused but this is not the question.

Ah, I didn't catch that! That explains a lot :)